wfe2/verify_test.go

```diff
@@ -28,6 +28,24 @@ import (
 	"gopkg.in/square/go-jose.v2"
 )
 
+// nonceServiceAdapter changes the gRPC nonce service interface to the one
+// required by jose. Used only for tests.
+type nonceServiceAdapter struct {
+	noncepb.NonceServiceClient
+}
+
+func (nsa nonceServiceAdapter) Nonce() (string, error) {
+	resp, err := nsa.NonceServiceClient.Nonce(context.Background(), &emptypb.Empty{})
+	if err != nil {
+		return "", err
+	}
+	return resp.Nonce, nil
+}
+
+func nonceService(wfe *WebFrontEndImpl) jose.NonceSource {
```

Comment:
I'm not hugely in love with calling this to construct a new nonceServiceAdapter every time jose needs one, even though it is just a thin wrapper around a pointer. What about having `setupWFE` return a nonceServiceAdapter that can then be passed into all of the various jose calls? Feel free to leave or take that option.

Reply:
Alright! This was a big change but the code is much nicer for it: https://github.com/letsencrypt/boulder/pull/6533/commits/a9b660384d8e28c1cc7295a52d22ef9b48e216e0

Now `setupWFE` returns a `requestSigner`, which can be used to sign all kinds of JWS's. This eliminates the repetitive `nonceService(&wfe)` parameter we had, and also the repetititive `t` parameter.